File: Fracto Project/Fracto.Fapi/src/app/models/appointment.ts
import { Doctor } from "./doctor";

export interface Appointment {
  appointmentId: number;
  doctorId: number;
  patientName: string;
  userId: number;
  appointmentDate: string;
  timeSlot?: string;   
  status?: string;     
  doctor?: Doctor;
}

export interface AppointmentCreateDto {
  doctorId: number;
  appointmentDate: string; 
  timeSlot: string;
  status?: string; 
}

File: Fracto Project/Fracto.API/Models/Appointment.cs
namespace Fracto.Models;
public class Appointment
{
    public int AppointmentId { get; set; }
    public int UserId { get; set; }
    public int DoctorId { get; set; }
    public DateTime AppointmentDate { get; set; }
    public string TimeSlot { get; set; } = string.Empty;
    public string Status { get; set; } = "Pending";
    public User? User { get; set; }
    public Doctor? Doctor { get; set; }
}